draft-ietf-jsonpath-base.md

```diff
@@ -1104,6 +1094,23 @@ Some examples:
 | `[3,4,5]` | `$[?index(@)==2]`<br>`$[?index(@)==17]` | `[5]`<br>`[]` | Select array element |
 | `{"a":{"b":{5},c:0}}` | `$[?@.b==5 && !@.c]` | `[{"b":{5},c:0}]` | Existence  |
 
+
+## Normalized Paths
+
+A Normalized Path identifies a node by providing a query that results in exactly that node. For example,
+the JSONPath expression `$.book[?(@.price<10)]` could select two values with Normalized Paths
+`$.book[3]` and `$.book[5]`.
+
+An example application of Normalized Paths is a JSONPath implementation which outputs Normalized Paths
+instead of, or in addition to, the values identified by these paths.
+
+Since bracket notation is more general than dot notation, it is convenient in constructing
+Normalized Paths. However, Normalized Paths are not required to use bracket notation when dot notation is appropriate.
+
+Normalized Paths are not necessarily canonical since two distinct Normalized Paths may identify the same node.
+For example, `$.a` and `$['a']` are equivalent Normalized Paths. With certain JSON documents, `$[-1]` and `$[4]` may
+be equivalent Normalized Paths.
```

Comment:
I'm not sure I agree with `$[-1]` being considered normalized.  In my opinion, a normalized path should be one that distinctly identifies a node regardless of the input JSON.  `$[-1]` will identify different nodes depending on the input.

Reply:
> I'm not sure I agree with `$[-1]` being considered normalized. In my opinion, a normalized path should be one that distinctly identifies a node regardless of the input JSON. `$[-1]` will identify different nodes depending on the input.

I have some reservations too, but I was trying to make clear one consequence of _not_ insisting on a canonical form of Normalized Path. However, I think Normalized Paths only make sense in the context of a known/given JSON document. If the document varies, then the node identified by _any_ path can be different (e.g. just replace the identified node in one document with a distinct node).

Reply:
`$[-1]` is now not allowed as part of a Normalized Path.